pkg/reconciler/build/build.go

```diff
@@ -22,6 +22,17 @@ import (
 	"github.com/shipwright-io/build/pkg/validate"
 )
 
+// build a list of current validation types
+var validationTypes = [...]string{
```

Comment:
I think we need to align the order of validations here with the one on the Build controller. For example, if we have the wrong strategy and the wrong secret on a Build and on a one-off BuildRun, the error reported on the status sub resource will be different. One will point out to a wrong secret, the other to a wrong strategy, we want both to say the same.

Reply:
Good catch, I fixed the order to match the sequence.